### ISSUE TITLE
delete_all_searches task triggered a postgres deadlock

### ISSUE THREAD

**Issue by eliotjordan** [pulibrary/pulfalight#958]
```
[PULFALight/production] ActiveRecord::Deadlocked: PG::TRDeadlockDetected: ERROR:  deadlock detectedDETAIL:  Process 21506 waits for ShareLock on transaction 1703152495; blocked by process 21508.Process 21508 waits for ShareLock on transaction 1703152493; blocked by process 21506.HINT:  See server log for query details.CONTEXT:  while deleting tuple (8629,12) in relation "searches": DELETE FROM "searches" WHERE (created_at < '2021-12-30' AND user_id IS NULL)
```

## Backtrace

[View full backtrace and more info at honeybadger.io](https://app.honeybadger.io/projects/65922/faults/83367881)

**Comment by eliotjordan** [comment]
Not sure if this a one-time issue or is recurring. Research is needed.

**Comment by tpendragon** [comment]
Resolve this when we determine if this always happens, and if it does always happen find a way to prevent it from happening.

**Comment by tpendragon** [comment]
Maybe this task is being run on multiple boxes?

**Comment by tpendragon** [comment]
This is a cron task installed by princeton_ansible on all `blacklight_app` boxes: https://github.com/pulibrary/princeton_ansible/blob/main/roles/blacklight_app/tasks/main.yml

It's weird we're only seeing it for Pulfalight though, I'd expect it for Orangelight, pulmap, and LAE as well.

**Comment by hackartisan** [comment]
The ansible-installed task was not working on orangelight and when I went to run it manually it crashed because of memory problems. I reimplemented the cleanup so it didn't load all the users in memory before starting to delete them.

Relevant PRs:
- https://github.com/pulibrary/orangelight/pull/2457/files
- https://github.com/pulibrary/orangelight/pull/2484/files

I put this in place on orangelight and DSS. Following through on all the apps that use the ansible-installed task and then removing it would good, either by submitting an upstream PR or by touching all the apps directly.
